src/include/platform/PlatformManager.h

```diff
@@ -199,8 +200,12 @@ class PlatformManager
                                  AttributeList<app::Clusters::FixedLabel::Structs::LabelStruct::Type, kMaxFixedLabels> & labelList);
     CHIP_ERROR SetUserLabelList(EndpointId endpoint,
                                 AttributeList<app::Clusters::UserLabel::Structs::LabelStruct::Type, kMaxUserLabels> & labelList);
-    CHIP_ERROR GetUserLabelList(EndpointId endpoint,
-                                AttributeList<app::Clusters::UserLabel::Structs::LabelStruct::Type, kMaxUserLabels> & labelList);
+    CHIP_ERROR AppendUserLabelList(EndpointId endpoint, app::Clusters::UserLabel::Structs::LabelStruct::Type & label);
+    CHIP_ERROR
+    GetUserLabelList(EndpointId endpoint,
+                     std::function<CHIP_ERROR(
+                         const AttributeList<app::Clusters::UserLabel::Structs::LabelStruct::Type, kMaxUserLabels> & labelList)>
```

Comment:
Needs to document the lifetime of the thing passed to the callback.

Also needs to document the general setup (that the callback will be invoked exactly once, before this function returns, etc, etc).

That said, do we have to use std::function here?  That's ... not small.  It might be simpler to have this callback be a C function and a closure `void *` especially since the one consumer captures a single pointer/reference.

Reply:
c callback can not take the lambda function which capture AttributeValueEncoder [&aEncoder] as the parameter. Another option is template.

Before I proceed I think we need to decide if we handle  GetUserLabelList/SetUserLabelList with platform APIs like now or application delegate APIs which take AttributeValueEncoder as the only parameter and offload everything to application.

Reply:
Hence "and a closure `void*`"...

Reply:
Refactor to GetUserLabelList/SetUserLabelList  from PlatformMgr to DeviceInfoProvider with iteration pattern